components/net/fetch/methods.rs

```diff
@@ -602,20 +602,30 @@ fn should_block_nosniff(request: &Request, response: &Response) -> bool {
         javascript_mime_types.contains(mime_type)
     }
 
+    // Returns the same mime type with parameters like the charset.
+    #[inline]
+    fn paramless_mime_type(mime_type: &Mime) -> Mime {
+        match *mime_type {
+            Mime(ref top, ref sub, _) => {
+               Mime(top.clone(), sub.clone(), vec![])
+            }
+        }
+    }
+
     let text_css: Mime = mime!(Text / Css);
     // Assumes str::starts_with is equivalent to mime::TopLevel
     return match type_ {
         // Step 6
         Type::Script => {
             match content_type_header {
-                Some(&ContentType(ref mime_type)) => !is_javascript_mime_type(&mime_type),
+                Some(&ContentType(ref mime_type)) => !is_javascript_mime_type(&paramless_mime_type(&mime_type)),
```

Comment:
I wonder if we could change `is_javascript_mime_type`
The last line:
```rust
javascript_mime_types.contains(mime_type)
```
could be something like this (maybe do some destructuring if you dont like `.0` and `.1`):
```rust
javascript_mime_types.iter().any(|type| type.0 == mime_type.0 && type.1 == mime_type.1)
```
This and @KiChjang's suggestion would remove the need for `paramless_mime_type`